com.unity.render-pipelines.universal/Runtime/VFXGraph/Shaders/VFXCommon.hlsl

```diff
@@ -116,14 +116,25 @@ float3 GetWorldStereoOffset()
 
 float VFXSampleDepth(float4 posSS)
 {
-    return LoadSceneDepth(uint2(posSS.xy));
+    float2 screenUV = GetNormalizedScreenSpaceUV(posSS.xy);
+
+    // In URP, the depth texture is optional and could be 4x4 white texture, Load isn't appropriate in that case.
+    //float depth = LoadSceneDepth(screenUV * _ScreenParams.xy);
+    float depth = SampleSceneDepth(screenUV);
+
+    return depth;
 }
 
 float VFXLinearEyeDepth(float depth)
 {
     return LinearEyeDepth(depth, _ZBufferParams);
 }
 
+float VFXLinearEyeDepthOrthographic(float depth)
```

Comment:
Is there any reason why those methods are actually pipeline specific? I guess LinearDepthToEyeDepth is a URP specific implem of what we did in HDRP VFXCommon?

Reply:
Afaik, this helper doesn't exist in HDRP, but indeed, LinearDepthToEyeDepth could be in core and shared between pipelines, we can even use a generic code. I used the centralized function because it was available, I realized afterwards @ludovic-theobald was doing the same kind of fix.
See the implementation in URP : https://github.com/Unity-Technologies/Graphics/blob/b468fa4627d191f06200a0f33df3d9cddee2c3f9/com.unity.render-pipelines.universal/ShaderLibrary/ShaderVariablesFunctions.hlsl#L328 in URP
...
Actually, it isn't exactly the same equation for `UNITY_REVERSED_Z` :
https://github.com/Unity-Technologies/Graphics/blob/6d552a7a5993e8296cea815dea63ef6ce0bf00e6/com.unity.render-pipelines.high-definition/Runtime/VFXGraph/Shaders/VFXCommon.hlsl#L154

Reply:
Good point, I didn't realize it was equivalent code. Actually, these two function can be store in cross SRP implementation : `VFXLinearEyeDepth` & `VFXLinearEyeDepthOrthographic` (VFXSampleDepth has to be implemented differently in URP and HDRP)

It was specific only for built-in renderpipeline : see [here](https://github.com/Unity-Technologies/Graphics/blob/7.x.x/release/com.unity.visualeffectgraph/Shaders/RenderPipeline/Legacy/VFXCommon.hlsl#L96) (_ZBufferParams is common to URP/HDRP)

I will move this code out from render pipeline specific implementation, it isn't needed anymore ⏳

Reply:
Fix at https://github.com/Unity-Technologies/Graphics/pull/4733/commits/91e182aecbfadb0ce7a813a08435759b2d950e45
I will propose another PR to move https://github.com/Unity-Technologies/Graphics/blob/b468fa4627d191f06200a0f33df3d9cddee2c3f9/com.unity.render-pipelines.universal/ShaderLibrary/ShaderVariablesFunctions.hlsl#L328 into core.

Reply:
Tested in URP :
![image](https://user-images.githubusercontent.com/22494706/120616997-febf1080-c459-11eb-9ecd-c58b2e015733.png)

And HDRP :
![image](https://user-images.githubusercontent.com/22494706/120617029-05e61e80-c45a-11eb-93ae-2ae42014f3bf.png)